lib/browser/api/app.js

```diff
@@ -77,6 +77,24 @@ app.allowNTLMCredentialsForAllDomains = function (allow) {
   }
 }
 
+const makeSingleInstance = app.makeSingleInstance
+app.makeSingleInstance = function (callback) {
+  return makeSingleInstance.call(app, function (argv, ...args) {
+    if (process.platform === 'win32') {
+      // Delay the checking so the callback is run as quickly as we would expect
+      setImmediate(() => {
+        argv.forEach((arg) => {
+          const protocolMatch = arg.match(/^([\s\S]+):\/\/[\s\S]+/)
```

Comment:
Electron has an internal implementation of Event but of the top of my head that isn't exposed.  I'm still looking into something better than that object